gatsbyjs/gatsby
# chore(www): convert HomepageBlogPosts to function component

## Description

Convert the `<HomepageBlogPosts/>` to function component from class based component

**Note:** There is no change functionality wise, just the implementation is different.

### How to test

1. Go to homepage of Gatsby using the links provided below
2. Scroll down to the **The Gatsby blog** section on homepage(screenshots are provided below)
3. This should work same as before
4. The layout of blog posts should change when resizing the browser to mobile screen sizes

**Local URL:** http://localhost:8000/
**Production URL:** https://www.gatsbyjs.org/

### Screenshot(s)

#### Desktop
<img width="640" alt="homepage-blog-posts-desktop" src="https://user-images.githubusercontent.com/19193724/85196307-f9bf8a80-b2f6-11ea-91e9-46a77aa01ca2.png">

#### Mobile
![homepage-blog-posts-mobile](https://user-images.githubusercontent.com/19193724/85196309-fc21e480-b2f6-11ea-8f40-2f09ba7ab76f.png)
